src/toast/pipeline_tools/__init__.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) 2019 by the parties listed in the AUTHORS file.
+# All rights reserved.  Use of this source code is governed by
+# a BSD-style license that can be found in the LICENSE file.
+
+import numpy as np
+
+from ..timing import function_timer, Timer
+from ..utils import Logger, Environment
+
+from .atm import (
+    add_atmosphere_args,
+    simulate_atmosphere,
+    scale_atmosphere_by_frequency,
+    update_atmospheric_noise_weights,
+    get_focalplane_radius,
+)
+from .binning import add_binner_args, init_binner, apply_binner
+from .dipole import add_dipole_args, simulate_dipole
+from .dist import add_dist_args, get_comm, get_time_communicators
+from .export import add_tidas_args, output_tidas, add_spt3g_args, output_spt3g
+from .filters import (
+    add_polyfilter_args,
+    apply_polyfilter,
+    add_groundfilter_args,
+    apply_groundfilter,
+)
+from .gain import add_gainscrambler_args, scramble_gains
+from .madam import add_madam_args, setup_madam, apply_madam
+from .noise import add_noise_args, simulate_noise, get_analytic_noise
+from .pointing import add_pointing_args, expand_pointing, get_submaps
+from .sky_signal import (
+    add_sky_map_args,
+    add_pysm_args,
+    scan_sky_signal,
+    simulate_sky_signal,
+)
+from .sss import add_sss_args, simulate_sss
+from .todground import (
+    add_todground_args,
+    get_breaks,
+    load_schedule,
+    load_weather,
+    Site,
+    Telescope,
+    CES,
+)
+from .todsatellite import add_todsatellite_args
+
+
+def add_mc_args(parser):
+    """ Add Monte Carlo arguments
+    """
+    parser.add_argument(
+        "--MC-start",
+        required=False,
+        default=0,
+        type=np.int,
+        help="First Monte Carlo noise realization",
+    )
+    parser.add_argument(
+        "--MC-count",
+        required=False,
+        default=1,
+        type=np.int,
+        help="Number of Monte Carlo noise realizations",
+    )
+    return
+
+
+@function_timer
```

Comment:
what is the difference between the `@function_timer` decorator and the timers you explicitly instantiate inside?

Reply:
The function timers can be collected and output in the end. The explicit timers only report when asked.